src/agent/driver-test-runner.js

```diff
@@ -0,0 +1,237 @@
+/// <reference path="../data/types.js" />
+/// <reference path="../shared/types.js" />
+/// <reference path="types.js" />
+
+import { WebDriver, until, By } from 'selenium-webdriver';
+
+import { startJob } from '../shared/job.js';
+
+import { ATDriver, ATKey } from './at-driver.js';
+import { AgentMessage } from './messages.js';
+
+/**
+ * @module agent
+ */
+
+const AFTER_NAVIGATION_DELAY = 1000;
+const AFTER_KEYS_DELAY = 5000;
+const AFTER_RUN_TEST_SETUP_BUTTON_DELAY = 5000;
+const RUN_TEST_SETUP_BUTTON_TIMEOUT = 1000;
+
+export class DriverTestRunner {
+  /**
+   * @param {object} options
+   * @param {AriaATCIShared.BaseURL} options.baseUrl
+   * @param {AriaATCIAgent.Log} options.log
+   * @param {WebDriver} options.webDriver
+   * @param {ATDriver} options.atDriver
+   */
+  constructor({ baseUrl, log, webDriver, atDriver }) {
+    this.baseUrl = baseUrl;
+    this.log = log;
+    this.webDriver = webDriver;
+    this.atDriver = atDriver;
+  }
+
+  /**
+   * @param {URL} url
+   */
+  async openPage({ url, referencePage }) {
+    await this.log(AgentMessage.OPEN_PAGE, { url });
+    await this.webDriver.switchTo().defaultContent();
+    await this.webDriver.navigate().to(url.toString());
+
+    try {
+      const loaded = this.webDriver.wait(async webDriver => {
```

Comment:
> The promise created by wait will never be fulfilled because the "condition" function always resolves to `undefined`.

Ah yup, I think I removed the truthy value during some cleanup.

> I think we just want to use the result of `executeAsyncScript` directly.

I tried that and it did not seem to resolve. I don't know `wait` well enough but it works with the execute inside the wait. Maybe I misunderstood the results I got that time, I'll try it again.